src/System.Binary.Base64/System/Binary/Base64.cs

```diff
@@ -190,17 +192,78 @@ public static int EncodeInPlace(Span<byte> buffer, int sourceLength)
             return encodedLength;
         }
 
+        #endregion
+
+        #region Decode
+
+        public static int ComputeDecodedLength(ReadOnlySpan<byte> source)
+        {
+            ref byte srcBytes = ref source.DangerousGetPinnableReference();
+
+            int srcLength = source.Length;
+
+            if ((srcLength & 0x3) != 0) return -1;   // Length of source is not a multiple of 4.
```

Comment:
Should this throw instead of returning -1? Passing in a buffer of non multiple of 4 bytes to Decode will fail, correct?